src/cli/rustup_mode.rs

```diff
@@ -84,8 +85,33 @@ where
     callee(cfg, matches)
 }
 
+async fn deprecated_async<'a, FF, F, B, R>(
+    instead: &str,
+    cfg: &'a mut Cfg,
+    matches: B,
+    callee: FF,
```

Comment:
I would suggest writing this as `impl FnOnce(&'a mut Cfg, B) -> impl Future<Output = R> + 'a`, which seems easier to understand to me.